stdlib/universe/aggregate_window.go

```diff
@@ -0,0 +1,951 @@
+	ts2, count := a.count.Merge(prevT, nextT, prev[1:], next[1:], mem)
+	ts2.Release()
+	return ts1, []array.Interface{sum[0], count[0]}
+}
+
+func (a aggregateWindowMean) Compute(buffers []array.Interface, mem memory.Allocator) (flux.ColType, array.Interface) {
+	count := buffers[1].(*array.Int)
+	b := array.NewFloatBuilder(mem)
+	b.Resize(count.Len())
+
+	switch sum := buffers[0].(type) {
+	case *array.Float:
+		for i, n := 0, sum.Len(); i < n; i++ {
+			v := sum.Value(i) / float64(count.Value(i))
+			b.Append(v)
+		}
+	case *array.Int:
+		for i, n := 0, sum.Len(); i < n; i++ {
+			v := float64(sum.Value(i)) / float64(count.Value(i))
+			b.Append(v)
+		}
+	case *array.Uint:
+		for i, n := 0, sum.Len(); i < n; i++ {
+			v := float64(sum.Value(i)) / float64(count.Value(i))
+			b.Append(v)
+		}
+	default:
+		panic("unreachable")
+	}
+	count.Release()
+	buffers[0].Release()
+	return flux.TFloat, b.NewArray()
+}
+
+func (a aggregateWindowMean) AppendEmpty(b array.Builder) {
+	b.AppendNull()
+}
+
+type AggregateWindowRule struct{}
+
+func (a AggregateWindowRule) Name() string {
+	return "AggregateWindowRule"
+}
+
+func (a AggregateWindowRule) Pattern() plan.Pattern {
+	return plan.Pat(WindowKind,
+		plan.Pat(SchemaMutationKind,
+			plan.OneOf([]plan.ProcedureKind{MeanKind, SumKind, CountKind},
+				plan.Pat(WindowKind, plan.Any()))))
+}
+
+func (a AggregateWindowRule) Rewrite(ctx context.Context, node plan.Node) (plan.Node, bool, error) {
+	if !feature.OptimizeAggregateWindow().Enabled(ctx) {
+		return node, false, nil
+	}
+
+	windowInfSpec := node.ProcedureSpec().(*WindowProcedureSpec)
+	if !a.isValidWindowInfSpec(windowInfSpec) {
+		return node, false, nil
+	}
+
+	duplicateNode := node.Predecessors()[0]
+	duplicateSpec := duplicateNode.ProcedureSpec().(*SchemaMutationProcedureSpec)
+
+	useStart, ok := a.isValidDuplicateSpec(duplicateSpec)
+	if !ok {
+		return node, false, nil
+	}
+
+	aggregateNode := duplicateNode.Predecessors()[0]
+	aggregate, valueCol, ok := a.isValidAggregateSpec(aggregateNode.ProcedureSpec())
+	if !ok {
+		return node, false, nil
+	}
+
+	windowNode := aggregateNode.Predecessors()[0]
+	windowSpec := windowNode.ProcedureSpec().(*WindowProcedureSpec)
+	if !a.isValidWindowSpec(windowSpec) {
+		return node, false, nil
+	}
+	parentNode := windowNode.Predecessors()[0]
+
+	parentNode.ClearSuccessors()
+	newNode := plan.CreateUniquePhysicalNode(ctx, "aggregateWindow", &AggregateWindowProcedureSpec{
+		spec:      windowSpec,
+		aggregate: aggregate,
+		valueCol:  valueCol,
+		useStart:  useStart,
+	})
+	parentNode.AddSuccessors(newNode)
+	newNode.AddPredecessors(parentNode)
+	return newNode, true, nil
+}
+
+func (a AggregateWindowRule) isValidWindowInfSpec(spec *WindowProcedureSpec) bool {
+	return spec.TimeColumn == execute.DefaultTimeColLabel &&
+		spec.StartColumn == execute.DefaultStartColLabel &&
+		spec.StopColumn == execute.DefaultStopColLabel &&
+		!spec.CreateEmpty ||
+		spec.Window.Every == infinityVar.Duration()
```

Comment:
Should this have parentheses around `(!spec.CreateEmpty || spec.Window.Every == infinityVar.Duration())` to get the meaning you want?